_posts/2023-09-05-use-opensearch-benchmark-with-randomized-query-terms.md

```diff
@@ -0,0 +1,171 @@
+
+Create a file called `workload.py` with the contents below, in the same directory as `workload.json`. The `random_search_term` function selects random terms from either the English words corpus or the BGG corpus and injects them into a term query body. The `register` function registers the `random_search_term` function as the `search-term-source` parameter source. 
+
+```
+import random
+import logging
+
+# Load the source terms from the bgg_words and english_words files. To 
+# keep it simple, the files are structured as a single set of terms per line.
+english_words = None
+with open('english_words.txt', 'r') as f:
+    lines = f.readlines()
+    english_words = list(map(lambda x: x.lstrip().rstrip(), lines))
+
+
+bgg_words = None
+with open('bgg_words.txt', 'r') as f:
+    lines = f.readlines()
+    bgg_words = list(map(lambda x: x.lstrip().rstrip(), lines))
+
+
+# This function generates and returns a full query body. Be sure to match
+# the function signature and return structure as below.
+def random_search_term(track, params, **kwargs):
+    index_name = params.get("index")
+
+    # Choose either a single english term, or one of the query
+    # term sets from the BGG data.
+    collection = random.choice([english_words, bgg_words])
+    query = random.choice(collection)
+
+    # logging.log(logging.INFO, f'Query: {query}')
+
+    # Returns a basic term query against the description field. In most cases, for performance
+    # testing, you want to disable the request cache. This carries through any cache parameter
+    # specified in the workload.
+    return {
+        "body": {
+            "query": {
+                "term": {
+                    "description": query
+                }
+            }
+        },
+        "index": index_name,
+        "cache": params.get("cache", False)
+    }
+
+
+# This maps the random_search_term function to the search-term-source 
+# parameter source specified in workload.json
+def register(registry):
+    registry.register_param_source("search-term-source", random_search_term)
+```
+
+Run OpenSearch Benchmark with the command line above. If you want to validate that Benchmark is pulling in the terms correctly, uncomment the line that logs the query terms. You can also enable slow logging in your OpenSearch cluster. **Note**, logging each query, or enabling slow logs will materially affect your performance results, so be sure to turn them off for actual measurement.
+
+## Summary
+
+In this post, you learned how to use `opensearch-benchmark create-workload` to read data from a running cluster, and build a test framework for that data. With the test framework in place, you generalized the search operation to provide a realistic set of queries by using terms from the corpus and from the English language. You can generalize this method to pull in and randomize elements of your query to use OpenSearch Benchmark and get accurate results.
```

Comment:
Last sentence: How about "You can use OpenSearch Benchmark to get accurate results by generalizing this method and pulling in and randomizing elements of your query"?